### PR TITLE
[ONNX] Add additional_test_kwargs into test_fx_to_onnx_with_onnxruntime.py

### PR DESCRIPTION
**Pull request by titaiwangms**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #99434

1. Expand additional_test_inputs to include kwargs
2. Revisit and update tests status by adding ops
3. Disabling dtype -1 assignment avoids potential bugs
4. Expand input/output to accept buit-in type, but they are not dynamically captured by dynamo.export right now, and they would be added as constant input to op.targets.
5. Move run_test_with_fx_to_onnx_exporter_and_onnx_runtime to onnx_test_common.py

### Summary
🛠️🧪🚀

This pull request improves the ONNX export support for scalar types and some ATen operators in PyTorch. It updates the test framework, the input and output adapters, the function dispatcher and the ONNX script generator to handle these cases. It also fixes or removes some failing or outdated tests.

> _We defy the limits of the ONNX script_
> _We export the models with scalar and copy_
> _We filter and convert the kwargs of dtype_
> _We run the tests with FX and docstring_

### Walkthrough
*  Update the `_InputArgsType` type annotation and the `_run_test_with_fx_to_onnx_exporter_and_onnx_runtime` function signature and docstring to handle int, float and bool inputs for some ONNX operators ([link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbL44-R46), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbL144-R157), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbL155-R164), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbL162-R172), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbL201-R224), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-0795f54fd1f38cfbf2c4a863a4efc9f40f2ea020a2b1612605c361b8d8d35862L197-R199), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-0795f54fd1f38cfbf2c4a863a4efc9f40f2ea020a2b1612605c361b8d8d35862L291-R293))
* Update the `filter_incompatible_and_dtype_convert_kwargs` function to omit the `dtype` argument if it is None ([link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-cabc3e58713d6fe7ab764ade4f2692f6753402322a7b542397cad16fcc72cf4bL203-R205))
* Update the test cases in `test_fx_to_onnx_with_onnxruntime.py` to use the `input_kwargs` parameter as a mapping, to fix the format of the `additional_test_inputs` parameter, and to add or remove `xfail`, `skip_dynamic_fx_test` and `skip_min_ort_version` decorators as needed ([link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbL320-R336), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbL330-R353), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbL357-R380), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbL452-L470), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbL488-R486), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbL509-R510), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbR543), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbL559-R565), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbL578-R580), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbL597-R599), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbL611-R620), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbL636-R636), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbL656-R659), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbL672-R675), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbL691-R698), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbL709-R714), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbL732-R730), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbL752-R750), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbL773-R771), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbR797-R803), [link](https://github.com/pytorch/pytorch/pull/99434/files?diff=unified&w=0#diff-c8fa56eefd7f98fb4f9739d57df57f02ede77e28528133736010a6d06651ebcbL807-R816))

### CODE REVIEW

**Comment by titaiwangms** [comment]
copilot:all

File: test/onnx/test_fx_to_onnx_with_onnxruntime.py [hunk 1]
```diff
@@ -464,25 +466,6 @@ def forward(self, tensor_x: torch.Tensor):
             self, MNISTModel(), (tensor_x,)
         )
 
-    @pytorch_test_common.skip_min_ort_version(
-        reason="ORT doesn't support dynamic fx exporter yet making SegFault flaky test",
-        version="1.15",
-        dynamic_only=True,
-    )
-    # test single op with no kwargs
-    def test_sigmoid(self):
```
Comment (BowenBao) [inline test/onnx/test_fx_to_onnx_with_onnxruntime.py:458]: is this expected?

Comment (titaiwangms) [reply test/onnx/test_fx_to_onnx_with_onnxruntime.py:458]: Yes, we have `test_sigmoid_add` already. I recall this one was like step-by-step test when we were in early stage of developing atenlib. Besides, we will land fx exporter version of test_ops_consistency.py anyway.

**Comment by BowenBao** [comment]
> copilot:all

Has to be in the main pr description section

File: torch/onnx/_internal/exporter.py [hunk 2]
```diff
@@ -288,7 +290,7 @@ def model_proto(self) -> onnx.ModelProto:
     @_beartype.beartype
     def adapt_torch_inputs_to_onnx(
         self, *model_args, **model_kwargs
-    ) -> Sequence[torch.Tensor]:
+    ) -> Sequence[Union[torch.Tensor, int, float, bool]]:
```
Comment (titaiwangms) [inline torch/onnx/_internal/exporter.py:293]: Might revert back due to https://github.com/pytorch/pytorch/issues/99534

File: test/onnx/test_fx_to_onnx_with_onnxruntime.py [hunk 3]
```diff
@@ -649,11 +448,14 @@ def forward(self, x):
...
                 return x
 
-        _run_test_with_fx_to_onnx_exporter_and_onnx_runtime(
-            self, MutationModel(), (torch.randn(12),), has_mutation=True
+        self.run_test_with_fx_to_onnx_exporter_and_onnx_runtime(
+            MutationModel(), (torch.randn(12),), has_mutation=True
         )
 
-    @pytorch_test_common.xfail("TypeError: missing a required argument: 'end'")
+    # TODO(justinchuby): A known limitation in aten::arange support.
```
Comment (justinchuby) [inline test/onnx/test_fx_to_onnx_with_onnxruntime.py:455]: Did I create this todo?

File: test/onnx/test_fx_to_onnx_with_onnxruntime.py [hunk 4]
```diff
@@ -684,15 +485,14 @@ def forward(self, x):
...
+        self.run_test_with_fx_to_onnx_exporter_and_onnx_runtime(
             Model(),
             (x,),
-            additional_test_inputs=[(x2,)],
+            additional_test_inputs=[((x2,),)],
         )
 
     @pytorch_test_common.xfail(
-        "RuntimeError: Unknown call_function target: aten.copy.default"
+        "RuntimeError: Unknown call_function target: aten.lift_fresh_copy.default"
```
Comment (justinchuby) [inline test/onnx/test_fx_to_onnx_with_onnxruntime.py:495]: 👀 is this a new op in the family

**Review by justinchuby — APPROVED** [review]

File: test/onnx/onnx_test_common.py [hunk 5]
```diff
@@ -140,3 +154,199 @@ def _run_test(m, remained_onnx_input_idx, flatten=True, ignore_none=True):
             )
         if not is_model_script and not self.is_script:
             _run_test(model, tracing_remained_onnx_input_idx)
+
+    @_beartype.beartype
+    def run_test_with_fx_to_onnx_exporter_and_onnx_runtime(
```
Comment (justinchuby) [inline test/onnx/onnx_test_common.py:159]: Does this need to be a method in the test class or can it be a global function? Prefer a global function if it does not require states in self?

Comment (titaiwangms) [reply test/onnx/onnx_test_common.py:159]: hmmm thought about that, but the function requires test_suite attributes: op_level_debug and dynamic_shapes. We could move it globally, but will need to impot class TestFx... to annotate, which is cons.

**Comment by titaiwangms** [comment]
@BowenBao PTAL

File: test/onnx/onnx_test_common.py (hunk 5 above)
Comment (justinchuby) [reply test/onnx/onnx_test_common.py:159]: SG! This is good then imo

File: test/onnx/test_fx_to_onnx_with_onnxruntime.py [hunk 6]
```diff
@@ -317,30 +138,22 @@ def func(x, b=torch.tensor(1.0)):
...
-    # as float 8.0 not <class "builtins.NoneType"> or <protocol "torch.Tensor">.
-    # @unittest.expectedFailure
     @pytorch_test_common.xfail(
-        "beartype.roar.BeartypeCallHintReturnViolation: @beartyped "
-        "torch.onnx._internal.exporter.ExportOutput.adapt_torch_inputs_to_onnx() "
-        "return (tensor([[[ 1.5410, -0.2934]]]), 8.0) violates type hint "
-        "typing.Sequence[torch.Tensor], as tuple index 1 item float 8.0 not "
-        "instance of <protocol 'torch.Tensor'>."
+        "https://github.com/pytorch/pytorch/issues/99534"
+        "To make it work, convert the float argument into a 0d tensor"
```
Comment (titaiwangms) [inline test/onnx/test_fx_to_onnx_with_onnxruntime.py:156]: I wonder if we still want to keep this test. xfail seems fine.

**Pull request closed without merging**